cmd/paasta/main.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+func listPaastaCommands() (map[string]bool, error) {
+	// alternatively:
+	// find $(echo $PATH | tr ':' ' ') -maxdepth 1 -xtype f -perm /o+x -name paasta-*
+	cmd := exec.Command("/bin/bash", "-p", "-c", "compgen -A command paasta-")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+	scanner := bufio.NewScanner(bytes.NewReader(out.Bytes()))
+	cmds := map[string]bool{}
+	for scanner.Scan() {
+		cmds[scanner.Text()] = true
+	}
+	return cmds, nil
+}
+
+func main() {
+	var cmds map[string]bool = nil
+	var subcmd string
+	var args []string
+
+	if len(os.Args) > 1 {
+		var err error
+		cmds, err = listPaastaCommands()
+		if err != nil {
+			fmt.Printf("Error generating list of sub-commands: %s\n", err)
+			os.Exit(1)
+		}
+
+		cmd := fmt.Sprintf("paasta-%s", os.Args[1])
+		if ok, _ := cmds[cmd]; ok {
+			var err error
+			subcmd, err = exec.LookPath(cmd)
```

Comment:
nit: `subcmd` and `cmd` naming is a bit confusing TBH.  Maybe rename `subcmd` into `prog`, `path` or something like that?